staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -58,6 +66,20 @@ func getEventKey(event *v1.Event) string {
 		"")
 }
 
+// getSpamKey builds unique event key based on source, involvedObject
+func getSpamKey(event *v1.Event) string {
+	return strings.Join([]string{
```

Comment:
Rather than doing join here, you could create a SpamStruct that contains these and key off it.  Slightly faster allocation and lookup and avoids a string join which is a lot of copies.  Not a huge deal.